bundler/mod.ts

```diff
@@ -167,7 +174,7 @@ export class Bundler {
         const depMod = this.#app.getModule(dep.url)
         if (depMod !== null) {
           const s = `.bundling.js#${dep.url}@`
-          await this.compile(depMod, external)
+          await this.compile(depMod, [dep.url, external].flat())
```

Comment:
in dev mode, we use a map to record the module urls before it's going to be compiled. that can avoid loop import issue. here better not change the external list.